openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -92,6 +99,45 @@ class CreditRequirementApiTests(CreditApiTestBase):
     """
     Test Python API for credit requirements and eligibility.
     """
+    def _mock_ecommerce_courses_api(self, body, status=200):
+        """ Mock GET requests to the ecommerce course API endpoint. """
+        httpretty.reset()
+        httpretty.register_uri(
+            httpretty.GET, "{}/courses/{}/?include_products=1".format(TEST_API_URL, unicode(self.course_key)),
+            status=status,
+            body=json.dumps(body), content_type="application/json",
+        )
+
+    @httpretty.activate
+    @override_settings(ECOMMERCE_API_URL=TEST_API_URL, ECOMMERCE_API_SIGNING_KEY=TEST_API_SIGNING_KEY)
```

Comment:
move this settings override to class level as I am seeing it on almost all tests